llvm/lib/Passes/StandardInstrumentations.cpp

```diff
@@ -830,6 +831,182 @@ void PrintIRInstrumentation::registerCallbacks(
   }
 }
 
+void DumpIRInstrumentation::registerCallbacks(
+    PassInstrumentationCallbacks &PIC) {
+
+  if (!(shouldDumpBeforeSomePass() || shouldDumpAfterSomePass()))
+    return;
+
+  this->PIC = &PIC;
+
+  PIC.registerBeforeNonSkippedPassCallback(
+      [this](StringRef P, Any IR) { this->pushPass(P, IR); });
```

Comment:
Could we omit `this` when possible? (ditto for other similar occurrences in this file)

Reply:
I don't yet understand why, but `this` is not implicitly captured here, it fails to compile without an explicit capture.

Reply:
> I don't yet understand why, but `this` is not implicitly captured here, it fails to compile without an explicit capture.

Hmmm alright, then leave it as it for now I guess